File: test/reference-output/basicEnumeration.ts
import * as svt from "simple-validation-tools";

export enum StringValues {
    first = "first",
    second = "second",
    third = "Third",
    fourth = "Fourth",
}

export function isStringValues(value: unknown): value is StringValues {
    return [StringValues.first, StringValues.second, StringValues.third, StringValues.fourth].some((v) => v === value);
}

export function validateStringValues(value: unknown): svt.ValidationResult<StringValues> {
    return svt.validateOneOfLiterals<StringValues>(value, [StringValues.first, StringValues.second, StringValues.third, StringValues.fourth]);
}

export enum IntValues {
    first = 1,
    second = 2,
    third = 3,
    fourth = 4,
}

export function isIntValues(value: unknown): value is IntValues {
    return [IntValues.first, IntValues.second, IntValues.third, IntValues.fourth].some((v) => v === value);
}

export function validateIntValues(value: unknown): svt.ValidationResult<IntValues> {
    return svt.validateOneOfLiterals<IntValues>(value, [IntValues.first, IntValues.second, IntValues.third, IntValues.fourth]);
}

File: test/reference-output/basicEnumeration.kt
package org.gotynoOutput

import com.fasterxml.jackson.annotation.*
import com.fasterxml.jackson.module.kotlin.*
import com.fasterxml.jackson.databind.annotation.*
import com.fasterxml.jackson.databind.*
import com.fasterxml.jackson.core.*
import com.fasterxml.jackson.databind.deser.std.*
import java.text.ParseException
import java.math.BigInteger
import kotlinx.serialization.Serializable
import org.gotynoDeclarations.BigIntegerSerializer

class BasicEnumeration {
enum class StringValues(val data: String) : java.io.Serializable {
    @JsonProperty("first") FIRST("first"),
    @JsonProperty("second") SECOND("second"),
    @JsonProperty("Third") THIRD("Third"),
    @JsonProperty("Fourth") FOURTH("Fourth");

    companion object {}
}

enum class IntValues(val data: Int) : java.io.Serializable {
    @JsonProperty(1) FIRST(1),
    @JsonProperty(2) SECOND(2),
    @JsonProperty(3) THIRD(3),
    @JsonProperty(4) FOURTH(4);

    companion object {}
}
}
